test/evergreen/print_stack_trace.py

```diff
@@ -109,49 +109,53 @@ def dump(self, exe_path: str, core_path: str, lib_path: str, dump_all: bool, out
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('-e', '--executable_path',
-                        help='path to the executable',
-                        required=True)
+                        help='path to the executable')
     parser.add_argument('-c', '--core_path',
-                        help='directory path to the core dumps',
-                        required=True)
+                        help='directory path to the core dumps')
     parser.add_argument('-l', '--lib_path', help='library path')
-    group = parser.add_mutually_exclusive_group(required=True)
-    group.add_argument('--unit_test', action='store_true', help='Format core dumps from python unit tests')
-    group.add_argument('--format', action='store_true', help='Format core dumps from format tests')
     args = parser.parse_args()
 
     # Store the path of the core files as a list.
     core_files = []
-    dump_all = args.unit_test
+    regex = re.compile(r'dump.*core', re.IGNORECASE)
+    core_path = args.core_path
+    if core_path is None:
+        core_path = '.'
 
-    regex = None
-    if (args.format):
-        regex = re.compile(r'dump_t.*core', re.IGNORECASE)
-    elif (args.unit_test):
-        regex = re.compile(r'dump.*python.*core', re.IGNORECASE)
-
-    for root, _, files in os.walk(args.core_path):
+    for root, _, files in os.walk(core_path):
         for file in files:
             if regex.match(file):
                 core_files.append(os.path.join(root, file))
 
     for core_file_path in core_files:
         print(border_msg(core_file_path), flush=True)
+
+        # If not specified, derive the binary name from the core dump folder.
+        if args.executable_path is None:
+            dirname = core_file_path.rsplit('/', 1)[0]
+            if 'cppsuite' in core_file_path:
+                executable_path = dirname + "/run"
+            elif 'format' in core_file_path:
+                executable_path = dirname + "/t"
+            else:
```

Comment:
Is there a pattern we can use instead of the catch-all `else` branch? If so, I'd prefer to use it. I'm worried that we're going to mysteriously fail to get core dumps if this directory doesn't exist. (If there's no pattern, we should probably log what we looked for, so it's clear why we didn't find something.)

Reply:
Alternatively, we already assume core files have a particular pattern on line 120. The binary name should be part of this, so we can probably extract it from there.

Reply:
As discussed offline, I used the group matching approach in this commit: b00af378d012ed8f280feb2f29d8a073ecf1317f
Let me know what you think

Reply:
Have you tried printing the captured binary name? My gut feeling is that this doesn't work as-is since the regex is greedy and will match until the very last `.` character, so the captured string will include the PID. If this turns out to be the case, then I think something like `dump_([^\.]*).*core` might work (i.e. the capture group is everything after the underscore until you reach a literal fullstop).

Reply:
Your are correct, it matches until the very last `.` which means it includes the PID, that's why I stripped it out with `rsplit`:
```
executable_name = result.group(1).rsplit('.', 1)[0]
```
I should have added a comment to make it easier to spot. I will try your regex, look good too.

Reply:
We found out that the assumption I had that the core would follow a pattern is wrong. I got into the following scenario:
```
dump_test_wt2853_per.853.core  test_wt2853_perf
```
It is not possible the retrieve the executable name from the core dump, the `f` is missing. I will revert to the original solution: 8c004b270b4ca36f646aed9c2ed535f2250bd494

Reply:
Updated in cb5955d9310127374ca5d950291154cbcf304d16